spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/StreamListenerMethodTests.java

```diff
@@ -0,0 +1,190 @@
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.config;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.Input;
+import org.springframework.cloud.stream.annotation.Output;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.MessagingException;
+import org.springframework.messaging.SubscribableChannel;
+
+/**
+ * @author Marius Bogoevici
+ * @author Ilayaperumal Gopinathan
+ */
+public class StreamListenerMethodTests {
+
+
+	@Test
+	public void testMethodWithInputAsMethodAndParameter() throws Exception {
+		try {
+			SpringApplication.run(TestMethodWithInputAsMethodAndParameter.class, "--server.port=0");
+			fail("Exception expected on using Input element as method and parameter");
+		}
+		catch (BeanCreationException e) {
+			assertThat(e.getCause().getMessage()).contains("@Input annotation can not be used in multiple method parameters");
+		}
+	}
+
+	@Test
+	public void testMethodWithOutputAsMethodAndParameter() throws Exception {
+		try {
+			SpringApplication.run(TestMethodWithOutputAsMethodAndParameter.class, "--server.port=0");
+			fail("Exception expected on using Input element as method and parameter");
+		}
+		catch (BeanCreationException e) {
+			assertThat(e.getCause().getMessage()).startsWith("Cannot set both Output (@Output/@SendTo) method annotation value ");
+		}
+	}
+
+	@Test
+	public void testMethodWithoutInput() throws Exception {
+		try {
+			SpringApplication.run(TestMethodWithoutInput.class, "--server.port=0");
+			fail("Exception expected when inbound target is not set");
+		}
+		catch (BeanCreationException e) {
+			assertThat(e.getCause().getMessage()).startsWith("Either StreamListener or a method parameter should be set with ");
+		}
+	}
+
+	@Test
+	public void testMethodWithMultipleInputParameters() throws Exception {
+		try {
+			SpringApplication.run(TestMethodWithMultipleInputParameters.class, "--server.port=0");
+			fail("Exception expected on multiple @Input method parameters");
+		}
+		catch (BeanCreationException e) {
+			assertThat(e.getCause().getMessage()).startsWith("@Input annotation can not be used in multiple method parameters");
+		}
+	}
+
+	@Test
+	public void testMethodWithMultipleOutputParameters() throws Exception {
```

Comment:
Another wrong test. A method using `@Input`/`@Output` can have an arbitrary number of inputs and outputs. What prompted this change?